index.js

```diff
@@ -109,24 +117,29 @@ const connect = async () => {
             });
         });
 
+        await producer.connect();
+
         return producer;
     } catch (err) {
-        logger.error('kafka broker connection failiure', err);
+        logger.error('kafka broker connection failure', err);
         throw err;
     }
 };
 
 /**
- * sends a message to a kafka topic
+ * sends a message to a kafka topic,
+ * call the connect method to get producer obj
+ * @param {Object} producer the producer object
  * @param {Object} data the data object
  * @param {String} topic the topic name
  */
-const sendMessage = async (data, topic) => {
+const sendMessage = async (producer, data, topic) => {
     const { job, buildConfig } = data;
     // after the produce has connected, we start start sending messages
 
     try {
         logger.info(`publishing msg ${job}:${buildConfig.buildId} to kafka topic:${topic}`);
+        await producer.connect();
```

Comment:
Looks like connection is established [when producer is created](https://github.com/screwdriver-cd/aws-producer-service/pull/6/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R120). Do we need to establish the connection again before sending a message?
```suggestion
```

Reply:
yes, that's the bug, its not a connection, producer needs to call a connect method before sending messages